src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -124,7 +124,7 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
 
         @Override public boolean start() throws Exception {
             EnvVars envReduced = new EnvVars(env);
-            EnvVars envHost = computer.getEnvironment();
+            EnvVars envHost = computer.buildEnvironment(TaskListener.NULL);
```

Comment:
I think I get the point.

The actual issue is `dockerClient.getVolumes` is ran passing `envHost`, which I guess is fine when used as for substracting host specific env, but prevent to customize node for DOCKER_HOST. Looking for a better fix